src/react/DeviceStatusPage/DeviceLiveEventsCard.tsx

```diff
@@ -0,0 +1,347 @@
+        });
+
+      astarte
+        .registerVolatileTrigger(roomName, disconnectionTriggerPayload)
+        .then(() => {
+          onInfoMessage('Watching for device disconnection events');
+        })
+        .catch(() => {
+          onErrorMessage("Coulnd't watch for device disconnection events");
+        });
+
+      astarte
+        .registerVolatileTrigger(roomName, errorTriggerPayload)
+        .then(() => {
+          onInfoMessage('Watching for device error events');
+        })
+        .catch(() => {
+          onErrorMessage("Coulnd't watch for device error events");
+        });
+
+      astarte
+        .registerVolatileTrigger(roomName, dataTriggerPayload)
+        .then(() => {
+          onInfoMessage('Watching for device data events');
+        })
+        .catch(() => {
+          onErrorMessage("Coulnd't watch for device data events");
+        });
+    })
+    .catch(() => {
+      onErrorMessage(`Couldn't join device ${deviceId} room`);
+    });
+}
+
+interface TimestampProps {
+  children: Date;
+}
+
+const Timestamp = ({ children }: TimestampProps): React.ReactElement => {
+  const formattedTimestamp = children.toISOString().substring(11, 23);
+
+  return <small className="text-secondary text-monospace mr-2">{`[${formattedTimestamp}]`}</small>;
+};
+
+const astarteDeviceEventBody = (event : AstarteDeviceEvent) => {
+  if (event instanceof AstarteDeviceConnectedEvent) {
+    return (
+      <>
+        <Badge variant="success" className="mr-2">
+          device connected
+        </Badge>
+        <span>IP : {event.ip}</span>
+      </>
+    );
+  }
+  if (event instanceof AstarteDeviceDisconnectedEvent) {
+    return (
+      <>
+        <Badge variant="warning" className="mr-2">
+          device disconnected
+        </Badge>
+        <span>Device disconnected</span>
+      </>
+    );
+  }
+  if (event instanceof AstarteDeviceIncomingDataEvent) {
+    return (
+      <>
+        <Badge variant="info" className="mr-2">
+          incoming data
+        </Badge>
+        <span className="mr-2">{event.interfaceName}</span>
+        <span className="mr-2">{event.path}</span>
+        <span className="mr-2 text-monospace">{event.value}</span>
+      </>
+    )
+  }
+  return <></>;
+};
+
+interface AstarteDeviceEventDelegateProps {
+  event: AstarteDeviceEvent;
+}
+
+const AstarteDeviceEventDelegate = ({ event }: AstarteDeviceEventDelegateProps) => {
+  return (
+    <li className="event-device px-2">
+      <Timestamp>{new Date(event.timestamp)}</Timestamp>
+      { astarteDeviceEventBody(event) }
+    </li>
+  );
+};
+
+interface SystemEventDelegateProps {
+  event: SystemEvent;
+}
+
+const SystemEventDelegate = ({ event }: SystemEventDelegateProps) => {
+  switch (event.level) {
+    case 'error':
+      return (
+        <li className="px-2">
+          <Timestamp>{new Date(event.timestamp)}</Timestamp>
+          <Badge variant="secondary" className="mr-2">
+            channel
+          </Badge>
```

Comment:
Maybe you could extract into a reusable component this `<Timestamp /><Badge />{children}` pattern and use it in all DeviceEvents and SystemEvents

Reply:
done that already, you'll see the change on the next push